doc/srv6/srv6_path_tracing_midpoint.md

```diff
@@ -0,0 +1,438 @@
+
+(2) The SONiC CLI uses the _portconfig_ script to inject the PT Interface ID and PT Timestamp Template into the PORT table in CONFIG_DB.
+
+(3) portmgrd receives a PORT update notification (portmgrd is an CONFIG_DB subscriber).
+
+(4) portmgrd inject the PT Interface ID and PT Timestamp Template into the PORT_TABLE in APPL_DB.
+
+(5) PortsOrch receives a PORT_TABLE update notification (PortsOrch is an APPL_DB subscriber).
+
+(6) PortsOrch invokes the sairedis `set_port_attribute()` APIs to set the port PT attributes including PT Interface ID, PT Timestamp Template, PT TAM Object. The port PT attributes are injected  into the ÀSIC_DB.
+
+The next subsections describe the changes required to support the PT Midpoint HLD described above. 
+
+### SONiC CLI Changes
+
+#### CLI Configuration Commands
+
+The existing `config interface` command is extended by introducing a new subcommand `pt` that allows users to configure the Interface ID and Timestamp Template parameters required for the PT Midpoint functionality. The Timestamp Template is optional. The default template is *template3*.
+
+```
+admin@sonic:~# config interface pt <interface_name> <pt_interface_id> [pt_timestamp_template]
+
+    <interface_name>: interface name 
+    <pt_interface_id>: Path Tracing Interface ID (0-4095)
+    [pt_timestamp_template]: Path Tracing timestamp template (optional)
+        Supported templates {template1, template2, template3, template4}
+            template1: timestamp[08:15]
+            template2: timestamp[12:19]
+            template3: timestamp[16:23]
+            template4: timestamp[20:27]
+        Default: template3
+```
+
+**Example 1:**
+
+```
+admin@sonic:~# config interface pt Ethernet8 128
+```
+
+The above command assigns Ethernet8 a Path Tracing Interface ID of 128 and a Timestamp Template template3 (default).
+
+**Example 2:**
+
+```
+admin@sonic:~# config interface pt Ethernet9 129 template2
+```
+
+The above command assigns Ethernet9 a Path Tracing Interface ID of 129 and a Timestamp Template template2.
+
+#### CLI Show Commands
+
+The existing `show interfaces` command is extended by introducing a new subcommand `pt` that allows users to verify the current PT Midpoint configuration (i.e., the Interface ID and Timestamp Template parameters).
+
+```
+admin@sonic:~$ show interfaces pt [interface_name]
+```
+
+**Example 1** - Show PT Midpoint configuration for all interfaces:
+
+```
+admin@sonic:~$ show interfaces pt
+
+      Interface            Alias    Oper    Admin    PT Interface ID   PT Timestamp Template
+---------------  ---------------  ------  -------  -----------------  ---------------------- 
+      Ethernet8   fortyGigE0/0/8      up       up                128               template3
+      Ethernet9   fortyGigE0/0/9      up       up                129               template2 
+```
+
+**Example 2** - Show PT Midpoint configuration for interface Ethernet8:
```

Comment:
CLI improvements - 
pt recording happens at the egress interface. - config interface path-tracing egress <interface_name> <pt_id> --template template2

Reply:
We are extending the CLI based on the feedback we got during the call to address the following: 
- have an explicit add/del
- use path-tracing instead of pt 

The updated CLI will be as follows 

- Disable PT on the interface 
`Config interface path-tracing del Ethernet8
`
- Enable PT on the interface and set its parameters 

`
Config interface path-tracing add Ethernet8 –interface-id 128 –ts-template template2
`